### ISSUE TITLE
Remove unnecessary null pointer checks

### ISSUE THREAD

**Issue by elfring** [tgfrerer/island#48]
[An extra null pointer check is not needed in functions](https://isocpp.org/wiki/faq/freestore-mgmt#delete-handles-null "Do I need to check for null before delete p?") like the following.
- [compute_example_app_destroy](https://github.com/tgfrerer/island/blob/9a5308831ed204d819129344bca15d6e5654192b/apps/examples/compute_example/compute_example_app/compute_example_app.cpp#L416 "compute_example_app_destroy function")
- [le_compute_pipeline_builder_destroy](https://github.com/tgfrerer/island/blob/427c66f2d770cb3af3a515ff6f7d4a860e43d24e/modules/le_pipeline_builder/le_pipeline_builder.cpp#L162 "le_compute_pipeline_builder_destroy function")

**Comment by tgfrerer** [comment]
Oh, true, good point! 

The check should not be for the `gpu_mesh` field but for the `self`, as we want to protect against accessing a field in an object that might have been already deleted...

e.g.

```
if ( self->gpu_mesh ) {
	delete self->gpu_mesh;
}
```
should be:

```
if ( self) {
	delete self->gpu_mesh;
}
```
(we still need to check for `self` being valid before possibly accessing its `gpu_mesh` field)

Would you like to send a PR? :)

**Comment by tgfrerer** [comment]
update: this should be fixed now as of 32a012b1f72909e6ff37486cc54a1b9227523872

**Comment by elfring** [comment]
:thought_balloon: Are you going to [reconsider any more implementation details](https://github.com/tgfrerer/island/blob/427c66f2d770cb3af3a515ff6f7d4a860e43d24e/modules/le_pipeline_builder/le_pipeline_builder.cpp#L162 "Update candidate: le_compute_pipeline_builder_destroy()")?
